--- a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi.org
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
@@ -212,9 +212,8 @@
 };
 
 &usb {
-	dr_mode = "peripheral";
+	dr_mode = "host";
 	extcon = <&pm8916_usbin>;
-
 	status = "okay";
 };
 
--- a/arch/arm64/boot/dts/qcom/msm8916-motorola-harpia.dts.org
+++ b/arch/arm64/boot/dts/qcom/msm8916-motorola-harpia.dts
@@ -264,6 +264,7 @@
 };
 
 &usb {
+	dr_mode = "host";
 	extcon = <&usb_id>, <&usb_id>;
 	status = "okay";
 };
