--- a/arch/arm64/boot/dts/qcom/sm7125-samsung-a52q.dts.org	2023-09-26 22:20:35.144168349 +0200
+++ b/arch/arm64/boot/dts/qcom/sm7125-samsung-a52q.dts	2023-09-26 22:18:36.691450133 +0200
@@ -843,7 +843,7 @@
 };
 
 &usb_1_dwc3 {
-	dr_mode = "peripheral";
+	dr_mode = "host";
 };
 
 &usb_1_hsphy {
--- a/arch/arm64/boot/dts/qcom/sm7125-samsung-a72q.dts.org	2023-09-26 22:20:29.928125452 +0200
+++ b/arch/arm64/boot/dts/qcom/sm7125-samsung-a72q.dts	2023-09-26 20:33:33.378971073 +0200
@@ -843,7 +843,7 @@
 };
 
 &usb_1_dwc3 {
-	dr_mode = "peripheral";
+	dr_mode = "host";
 };
 
 &usb_1_hsphy {
