diff --git a/Documentation/devicetree/bindings/interconnect/qcom,msm8998-bwmon.yaml b/Documentation/devicetree/bindings/interconnect/qcom,msm8998-bwmon.yaml
index 5d17bdcfd..b63db9098 100644
--- a/Documentation/devicetree/bindings/interconnect/qcom,msm8998-bwmon.yaml
+++ b/Documentation/devicetree/bindings/interconnect/qcom,msm8998-bwmon.yaml
@@ -25,6 +25,7 @@ properties:
       - const: qcom,msm8998-bwmon       # BWMON v4
       - items:
           - enum:
+              - qcom,sc7180-cpu-bwmon
               - qcom,sc7280-cpu-bwmon
               - qcom,sc8280xp-cpu-bwmon
               - qcom,sdm845-cpu-bwmon
@@ -32,6 +33,7 @@ properties:
           - const: qcom,sdm845-bwmon    # BWMON v4, unified register space
       - items:
           - enum:
+              - qcom,sc7180-llcc-bwmon
               - qcom,sc8280xp-llcc-bwmon
               - qcom,sm8550-llcc-bwmon
           - const: qcom,sc7280-llcc-bwmon
diff --git a/arch/arm64/boot/dts/qcom/pm6150.dtsi b/arch/arm64/boot/dts/qcom/pm6150.dtsi
index 2e6afa296..a66c36d1f 100644
--- a/arch/arm64/boot/dts/qcom/pm6150.dtsi
+++ b/arch/arm64/boot/dts/qcom/pm6150.dtsi
@@ -88,6 +88,12 @@ pm6150_adc_tm: adc-tm@3500 {
 			status = "disabled";
 		};
 
+		rtc@6000 {
+			compatible = "qcom,pm8941-rtc";
+			reg = <0x6000>;
+			interrupts = <0x0 0x61 0x1 IRQ_TYPE_NONE>;
+		};
+
 		pm6150_gpios: gpio@c000 {
 			compatible = "qcom,pm6150-gpio", "qcom,spmi-gpio";
 			reg = <0xc000>;
diff --git a/arch/arm64/boot/dts/qcom/sc7180-lite.dtsi b/arch/arm64/boot/dts/qcom/sc7180-lite.dtsi
index 4b306a59d..975d4422f 100644
--- a/arch/arm64/boot/dts/qcom/sc7180-lite.dtsi
+++ b/arch/arm64/boot/dts/qcom/sc7180-lite.dtsi
@@ -6,21 +6,21 @@
  */
 
 &cpu6_opp10 {
-	opp-peak-kBps = <7216000 22425600>;
+	opp-peak-kBps = <22425600>;
 };
 
 &cpu6_opp11 {
-	opp-peak-kBps = <7216000 22425600>;
+	opp-peak-kBps = <22425600>;
 };
 
 &cpu6_opp12 {
-	opp-peak-kBps = <8532000 23347200>;
+	opp-peak-kBps = <23347200>;
 };
 
 &cpu6_opp13 {
-	opp-peak-kBps = <8532000 23347200>;
+	opp-peak-kBps = <23347200>;
 };
 
 &cpu6_opp14 {
-	opp-peak-kBps = <8532000 23347200>;
+	opp-peak-kBps = <23347200>;
 };
diff --git a/arch/arm64/boot/dts/qcom/sc7180-samsung-galaxy-book-go.dts b/arch/arm64/boot/dts/qcom/sc7180-samsung-galaxy-book-go.dts
index 3ce174640..29272444d 100644
--- a/arch/arm64/boot/dts/qcom/sc7180-samsung-galaxy-book-go.dts
+++ b/arch/arm64/boot/dts/qcom/sc7180-samsung-galaxy-book-go.dts
@@ -4,7 +4,9 @@
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/regulator/qcom,rpmh-regulator.h>
+
 #include "sc7180.dtsi"
+
 #include "pm6150.dtsi"
 #include "pm6150l.dtsi"
 
@@ -13,13 +15,10 @@ / {
 	compatible = "samsung,galaxy-book-go", "qcom,sc7180";
 	chassis-type = "laptop";
 
-	chosen {
-		bootargs = "efi=novamap earlycon=efifb clk_ignore_unused pd_ignore_unused console=efifb";
-	};
-
+// lets assume they are the same on the galaxy book go like on the aspire 1 for now ...
 	reserved-memory {
 		zap_mem: zap-shader@8c515000 {
-			reg = <0x0 0x8c515000 0x0 0x2000>;
+			reg = <0 0x8c515000 0 0x2000>;
 			no-map;
 		};
 
@@ -29,7 +28,7 @@ mpss_mem: mpss@86000000 {
 		};
 
 		venus_mem: venus@8f600000 {
-			reg = <0x0 0x8f600000 0x0 0x500000>;
+			reg = <0 0x8f600000 0 0x500000>;
 			no-map;
 		};
 
@@ -48,31 +47,633 @@ adsp_mem: adsp@90800000 {
 			no-map;
 		};
 	};
+
+	aliases {
+// LATER - this is bluetooth - lets do this later ...
+//		bluetooth0 = &bluetooth;
+//		hsuart0 = &uart3;
+		serial0 = &uart8;
+		wifi0 = &wifi;
+	};
+
+// NEW - touchpad - based on aceridus work
+	vdd_touchpad: regulator-vdd-touchpad {
+		compatible = "regulator-fixed";
+		regulator-name = "vdd-touchpad";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+// 		gpio = <&tlmm 31 GPIO_ACTIVE_HIGH>;
+// CHANGED - above value was used so far, but the galaxy dsl says 0x20 = 32 in TPXC
+		gpio = <&tlmm 32 GPIO_ACTIVE_HIGH>;
+// CHANGED - by looking at the c630 dsl and dts it looks like TSC1 looks similar: irq and gpio in dts is 125
+//           and that is only the first TLMMGPIO in dsl = 0x7D / 0x007D for the different TSC1 entries
+//           that first TLMMGPIO there has all Zero and maybe due to this has IRQ_TYPE_LEVEL_LOW maybe (LABEL_01 - for a note below)
+//		gpio = <&tlmm 31 GPIO_ACTIVE_LOW>;
+// 		/* add "enable-active-high;" when GPIO is active high */
+ 		enable-active-high;
+		pinctrl-names = "default";
+		pinctrl-0 = <&en_touchpad>;
+	};
+
+// OLD - touchpad
+//	reg_tp_3p3: touchpad-regulator {
+//		compatible = "regulator-fixed";
+//		regulator-name = "tp_3p3";
+//		regulator-min-microvolt = <3300000>;
+//		regulator-max-microvolt = <3300000>;
+//
+//		gpio = <&tlmm 25 GPIO_ACTIVE_HIGH>;
+//		enable-active-high;
+//
+//		pinctrl-names = "default";
+//		pinctrl-0 = <&reg_tp_en_default>;
+//	};
+
+// LATER - this is audio - lets do this later ...
+// 	reg_codec_3p3: codec-regulator {
+// 		compatible = "regulator-fixed";
+// 		regulator-name = "codec_3p3";
+// 		regulator-min-microvolt = <3300000>;
+// 		regulator-max-microvolt = <3300000>;
+// 
+// 		gpio = <&tlmm 83 GPIO_ACTIVE_HIGH>;
+// 		enable-active-high;
+// 
+// 		pinctrl-names = "default";
+// 		pinctrl-0 = <&reg_audio_en_default>;
+// 	};
+
+// 	reg_lcm_3p3: panel-regulator {
+// 		compatible = "regulator-fixed";
+// 		regulator-name = "lcm_3p3";
+// 		regulator-min-microvolt = <3300000>;
+// 		regulator-max-microvolt = <3300000>;
+
+// // INFO - comment from the acer dts
+// // 		/* HACK: Display fails with
+// // 		 *
+// // 		 * *ERROR* Unexpected max rate (0x0); assuming 5.4 GHz
+// // 		 * *ERROR* Link training failed, link is off (-5)
+// // 		 *
+// // 		 * if the power to the panel was cut
+// // 		 */
+// 		regulator-always-on;
+
+// TODO
+// 		gpio = <&tlmm 26 GPIO_ACTIVE_HIGH>;
+// 		enable-active-high;
+
+// 		pinctrl-names = "default";
+// 		pinctrl-0 = <&reg_lcm_en_default>;
+// 	};
+
+	reg_brij_1p8: bridge-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "brij_1p8";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+
+		vin-supply = <&vreg_l8c_1p8>;
+
+// TODO
+		gpio = <&tlmm 20 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&reg_edp_1p8_en_default>;
+	};
+
+	reg_brij_1p2: bridge-1p2-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "brij_1p2";
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+
+// TODO
+		gpio = <&tlmm 19 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&reg_edp_1p2_en_default>;
+	};
+
+// 	backlight: backlight {
+// 		compatible = "pwm-backlight";
+// 		pwms = <&sn65dsi86_bridge 1000000>;
+// // TODO
+// 		enable-gpios = <&tlmm 10 GPIO_ACTIVE_HIGH>;
+
+// 		pinctrl-names = "default";
+// 		pinctrl-0 = <&soc_bkoff_default>;
+// 	};
+
+// LATER - this is audio - lets do this later ...
+// 	max98357a: audio-codec-0 {
+// 		compatible = "maxim,max98357a";
+// 		pinctrl-names = "default";
+// 		pinctrl-0 = <&amp_sd_mode_default>;
+// 		sdmode-gpios = <&tlmm 23 GPIO_ACTIVE_HIGH>;
+// 		#sound-dai-cells = <0>;
+// 	};
+};
+
+// NEW - ufs storage basic definitions taken over from sm7125 are defined in sc7180.dtsi now in this tree
+
+&ufs_mem_hc {
+	status = "okay";
+
+	vcc-supply = <&vreg_l19a_2p9>;
+	vcc-max-microamp = <600000>;
+	vccq2-supply = <&vreg_l12a_1p8>;
+	vccq2-max-microamp = <600000>;
+};
+
+&ufs_mem_phy {
+	status = "okay";
+
+	vdda-phy-supply = <&vreg_l4a_0p8>;
+	vdda-phy-max-microamp = <62900>;
+	vdda-pll-supply = <&vreg_l3c_1p2>;
+	vdda-pll-max-microamp = <18300>;
+};
+
+&camcc { status = "disabled"; };
+
+&mdss_dsi0 {
+	status = "okay";
+	vdda-supply = <&vreg_l3c_1p2>;
+};
+
+&mdss_dsi0_out {
+	remote-endpoint = <&sn65dsi86_in>;
+	data-lanes = <0 1 2 3>;
+};
+
+&mdss_dsi0_phy {
+	status = "okay";
+	vdds-supply = <&vreg_l4a_0p8>;
+};
+
+&gpu {
+	status = "okay";
+	zap-shader {
+		memory-region = <&zap_mem>;
+		firmware-name = "qcom/sc7180-samsung-galaxy-book-go/qcdxkmsuc7180.mbn";
+	};
+};
+
+// NEW display bridge - it seems to be on i2c0 (I2C1 in galaxy dsl looks similar to IC11 = i2c10 in acer dsl
+// both acer and galaxy dsl have the same sn65 strings in them, so lets assume that it is using the same display bridge
+&i2c0 {
+	status = "okay";
+	clock-frequency = <400000>;
+
+	sn65dsi86_bridge: bridge@2c {
+		compatible = "ti,sn65dsi86";
+		reg = <0x2c>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&bridge_en_default>,
+			<&edp_bridge_irq_default>,
+			<&bridge_suspend_default>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		#pwm-cells = <1>;
+
+		interrupt-parent = <&tlmm>;
+// TODO
+		interrupts = <11 IRQ_TYPE_LEVEL_HIGH>;
+
+// TODO: acer has TLMGPIO 0x56 and 0x57 - galaxy has 0x22 and 0x23 - how to mape those?
+		enable-gpios = <&tlmm 51 GPIO_ACTIVE_HIGH>;
+		suspend-gpios = <&tlmm 22 GPIO_ACTIVE_LOW>;
+
+		vpll-supply = <&reg_brij_1p8>;
+		vccio-supply = <&reg_brij_1p8>;
+		vcca-supply = <&reg_brij_1p2>;
+		vcc-supply = <&reg_brij_1p2>;
+
+		clocks = <&rpmhcc RPMH_LN_BB_CLK3>;
+		clock-names = "refclk";
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				sn65dsi86_in: endpoint {
+					remote-endpoint = <&mdss_dsi0_out>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				sn65dsi86_out: endpoint {
+					data-lanes = <0 1>;
+					remote-endpoint = <&panel_in_edp>;
+				};
+			};
+		};
+
+		aux-bus {
+// INFO - searching for PanelNam (without the e at the end to avoid the line break in the search) gives "eDP dynamic"
+//        in the galaxy dsl where it gives a real panel name in the acer dsl "NV140FHM-N64"
+			panel: panel {
+				compatible = "edp-panel";
+// 				power-supply = <&reg_lcm_3p3>;
+// 				backlight = <&backlight>;
+
+				port {
+					panel_in_edp: endpoint {
+						remote-endpoint = <&sn65dsi86_out>;
+					};
+				};
+			};
+		};
+	};
+};
+
+// might be the same as I2C3 entry in acer and galaxy dsl look similar
+&i2c2 {
+	status = "okay";
+	clock-frequency = <400000>;
+	/* Embedded Controller in acer aspire 1*/
+	/* something with EMEC in galaxy dsl */
+};
+
+// OLD - keyboard and touchpad - replaced with new and different setup
+// &i2c4 {
+// 	status = "okay";
+// 	clock-frequency = <400000>;
+// 
+//	/*
+//	 * NOTE: DSDT defines two possible touchpads, other one is
+//	 *
+//	 * reg = <0x15>;
+//	 * hid-descr-addr = <0x1>;
+//	 */
+//
+//	touchpad@2c {
+//		compatible = "hid-over-i2c";
+//		reg = <0x2c>;
+//		hid-descr-addr = <0x20>;
+//
+//		vdd-supply = <&reg_tp_3p3>;
+//
+//		interrupts-extended = <&tlmm 94 IRQ_TYPE_LEVEL_LOW>;
+//
+//		pinctrl-names = "default";
+//		pinctrl-0 = <&hid_touchpad_default>;
+//	};
+// 
+// 	keyboard@3a {
+// 		compatible = "hid-over-i2c";
+// 		reg = <0x3a>;
+// 		hid-descr-addr = <0x1>;
+// 
+// 		interrupts-extended = <&tlmm 33 IRQ_TYPE_LEVEL_LOW>;
+// 
+// 		pinctrl-names = "default";
+// 		pinctrl-0 = <&hid_keyboard_default>;
+// 
+// 		wakeup-source;
+// 	};
+// };
+
+&i2c4 {
+	status = "okay";
+	clock-frequency = <400000>;
+	/* something with EMEC in dsl */
+};
+
+&i2c5 {
+	status = "okay";
+	clock-frequency = <400000>;
+	/* something with EMEC in dsl */
+};
+
+// NEW - keyboard - based on aceridus work
+&i2c6 {
+	clock-frequency = <400000>;
+	status = "okay";
+
+	keyboard@5 {
+		compatible = "hid-over-i2c";
+		reg = <0x5>;
+		hid-descr-addr = <0x20>;
+
+// 		interrupts-extended = <&tlmm 32 IRQ_TYPE_LEVEL_LOW>;
+// UPDATE - 32 is definitely wrong as its definitely defined as enable gpio for the touchpad
+//          experiments have shown 38 to work as well and as bad as 32, 30-42 tested so far
+		interrupts-extended = <&tlmm 38 IRQ_TYPE_LEVEL_LOW>;
+// CHANGED - I2C7 in galaxy dsl has 0x3B and 0X3C set which would be 59 and 60
+//		interrupts-extended = <&tlmm 59 IRQ_TYPE_LEVEL_LOW>;
+// UPDATE - this does not seem to work ...
+// TODO - maybe correct already as the keyboard works
+
+		wakeup-source;
+	};
+};
+
+&i2c7 {
+	status = "okay";
+	clock-frequency = <400000>;
+	/* something with EMEC in dsl */
+};
+
+// LATER - this is audio - lets do this later ...
+// &i2c9 {
+// 	status = "okay";
+// 	clock-frequency = <400000>;
+// 
+// 	alc5682: codec@1a {
+// 		compatible = "realtek,rt5682i";
+// 		reg = <0x1a>;
+// 		pinctrl-names = "default";
+// 		pinctrl-0 = <&codec_irq_default>;
+// 
+// 		#sound-dai-cells = <1>;
+// 
+// 		interrupt-parent = <&tlmm>;
+// 		/*
+// 		 * This will get ignored because the interrupt type
+// 		 * is set in rt5682.c.
+// 		 */
+// 		interrupts = <28 IRQ_TYPE_EDGE_BOTH>;
+// 
+// 		AVDD-supply = <&vreg_l15a_1p8>;
+// 		MICVDD-supply = <&reg_codec_3p3>;
+// 		VBAT-supply = <&reg_codec_3p3>;
+// 
+// 		realtek,dmic1-data-pin = <1>;
+// 		realtek,dmic1-clk-pin = <1>;
+// 		realtek,jd-src = <1>;
+// 	};
+// };
+
+// NEW touchpad - based on aceridus work
+&i2c10 {
+	clock-frequency = <400000>;
+	status = "okay";
+
+	touchpad@40 {
+		compatible = "hid-over-i2c";
+		reg = <0x40>;
+		hid-descr-addr = <0x0e>;
+
+// TODO - i think this was found out already as the proper one in the irc session
+		interrupts-extended = <&tlmm 33 IRQ_TYPE_LEVEL_LOW>;
+// CHANGED - maybe it is actually 31 or maybe 32 - see TPXC in dsl
+//		interrupts-extended = <&tlmm 31 IRQ_TYPE_LEVEL_LOW>;
+
+		vdd-supply = <&vdd_touchpad>;
+		vddl-supply = <&vreg_l10a_1p8>;
+		post-power-on-delay-ms = <150>;
+
+		wakeup-source;
+	};
+};
+
+// OLD display bridge - maybe needs to move to another i2c port?
+// &i2c10 {
+// 	status = "okay";
+// 	clock-frequency = <400000>;
+// 
+// 	sn65dsi86_bridge: bridge@2c {
+// 		compatible = "ti,sn65dsi86";
+// 		reg = <0x2c>;
+// 		pinctrl-names = "default";
+// 		pinctrl-0 = <&bridge_en_default>,
+// 			<&edp_bridge_irq_default>,
+// 			<&bridge_suspend_default>;
+// 		gpio-controller;
+// 		#gpio-cells = <2>;
+// 		#pwm-cells = <1>;
+// 
+// 		interrupt-parent = <&tlmm>;
+// 		interrupts = <11 IRQ_TYPE_LEVEL_HIGH>;
+// 
+// 		enable-gpios = <&tlmm 51 GPIO_ACTIVE_HIGH>;
+// 		suspend-gpios = <&tlmm 22 GPIO_ACTIVE_LOW>;
+// 
+// 		vpll-supply = <&reg_brij_1p8>;
+// 		vccio-supply = <&reg_brij_1p8>;
+// 		vcca-supply = <&reg_brij_1p2>;
+// 		vcc-supply = <&reg_brij_1p2>;
+// 
+// 		clocks = <&rpmhcc RPMH_LN_BB_CLK3>;
+// 		clock-names = "refclk";
+// 
+// 		ports {
+// 			#address-cells = <1>;
+// 			#size-cells = <0>;
+// 
+// 			port@0 {
+// 				reg = <0>;
+// 				sn65dsi86_in: endpoint {
+// 					remote-endpoint = <&mdss_dsi0_out>;
+// 				};
+// 			};
+// 
+// 			port@1 {
+// 				reg = <1>;
+// 				sn65dsi86_out: endpoint {
+// 					data-lanes = <0 1>;
+// 					remote-endpoint = <&panel_in_edp>;
+// 				};
+// 			};
+// 		};
+// 
+// 		aux-bus {
+// 			panel: panel {
+// 				compatible = "edp-panel";
+// 				power-supply = <&reg_lcm_3p3>;
+// 				backlight = <&backlight>;
+// 
+// 				port {
+// 					panel_in_edp: endpoint {
+// 						remote-endpoint = <&sn65dsi86_out>;
+// 					};
+// 				};
+// 			};
+// 		};
+// 	};
+// };
+
+// LATER - this seems to be audio too - so lets do this later as well ...
+// /* Seems like ADSP really insists on managing those itself */
+&lpasscc { status = "disabled"; };
+&lpass_hm { status = "disabled"; };
+
+&mdp {
+	status = "okay";
+};
+
+&mdss {
+	status = "okay";
+};
+
+&pm6150_adc {
+	thermistor@4e {
+		reg = <ADC5_AMUX_THM2_100K_PU>;
+		qcom,ratiometric;
+		qcom,hw-settle-time = <200>;
+	};
+
+	charger-thermistor@4f {
+		reg = <ADC5_AMUX_THM3_100K_PU>;
+		qcom,ratiometric;
+		qcom,hw-settle-time = <200>;
+	};
+};
+
+&pm6150_adc_tm {
+	status = "okay";
+
+	charger-thermistor@0 {
+		reg = <0>;
+		io-channels = <&pm6150_adc ADC5_AMUX_THM3_100K_PU>;
+		qcom,ratiometric;
+		qcom,hw-settle-time-us = <200>;
+	};
+
+	thermistor@1 {
+		reg = <1>;
+		io-channels = <&pm6150_adc ADC5_AMUX_THM2_100K_PU>;
+		qcom,ratiometric;
+		qcom,hw-settle-time-us = <200>;
+	};
+};
+
+&pm6150_pon { status = "disabled"; };
+
+&qupv3_id_0 {
+	status = "okay";
+};
+
+&qupv3_id_1 {
+	status = "okay";
+};
+
+&remoteproc_mpss {
+	status = "okay";
+	firmware-name = "qcom/sc7180-samsung-galaxy-book-go/qcmpss7180_nm.mbn";
+// TODO - maybe the firmware with modem support is required due to the builtin modem?
+//	firmware-name = "qcom/sc7180-samsung-galaxy-book-go/qcmpss7180_BTU.mbn";
+};
+
+// OLD - sdhc_2 worked for the galaxy, so lets use that one
+// &sdhc_1 {
+// 	status = "okay";
+// 
+// 	pinctrl-names = "default", "sleep";
+// 	pinctrl-0 = <&sdc1_default>;
+// 	pinctrl-1 = <&sdc1_sleep>;
+// 	vmmc-supply = <&vreg_l19a_2p9>;
+// 	vqmmc-supply = <&vreg_l12a_1p8>;
+// };
+
+// NEW - sdhc_2 from aceridus adjusted to pinctrl and supplies from this dts
+&sdhc_2 {
+	status = "okay";
+
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc2_on>;
+	pinctrl-1 = <&sdc2_off>;
+	vmmc-supply = <&vreg_l9c_2p9>;
+	vqmmc-supply = <&vreg_l6c_2p9>;
+
+	cd-gpios = <&tlmm 69 GPIO_ACTIVE_LOW>;
+};
+
+// LATER - this is bluetooth - lets do this later ...
+// &uart3 {
+// 	status = "okay";
+// 
+// 	/delete-property/interrupts;
+// 	interrupts-extended = <&intc GIC_SPI 604 IRQ_TYPE_LEVEL_HIGH>,
+// 				<&tlmm 41 IRQ_TYPE_EDGE_FALLING>;
+// 
+// 	pinctrl-names = "default", "sleep";
+// 	pinctrl-1 = <&qup_uart3_sleep>;
+// 
+// 	bluetooth: bluetooth {
+// 		compatible = "qcom,wcn3991-bt";
+// 		vddio-supply = <&vreg_l10a_1p8>;
+// 		vddxo-supply = <&vreg_l1c_1p8>;
+// 		vddrf-supply = <&vreg_l2c_1p3>;
+// 		vddch0-supply = <&vreg_l10c_3p3>;
+// 		max-speed = <3200000>;
+// 	};
+// };
+
+&uart8 {
+	status = "okay";
+};
+
+&usb_1 {
+	status = "okay";
+};
+
+&usb_1_dwc3 {
+	dr_mode = "host";
+};
+
+&usb_1_hsphy {
+	status = "okay";
+	vdd-supply = <&vreg_l4a_0p8>;
+	vdda-pll-supply = <&vreg_l11a_1p8>;
+	vdda-phy-dpdm-supply = <&vreg_l17a_3p0>;
+	qcom,imp-res-offset-value = <8>;
+	qcom,preemphasis-level = <QUSB2_V2_PREEMPHASIS_15_PERCENT>;
+	qcom,preemphasis-width = <QUSB2_V2_PREEMPHASIS_WIDTH_HALF_BIT>;
+	qcom,bias-ctrl-value = <0x22>;
+	qcom,charge-ctrl-value = <3>;
+	qcom,hsdisc-trim-value = <0>;
+};
+
+&usb_1_qmpphy {
+	status = "okay";
+	vdda-phy-supply = <&vreg_l3c_1p2>;
+	vdda-pll-supply = <&vreg_l4a_0p8>;
+};
+
+&venus {
+	firmware-name = "qcom/sc7180-samsung-galaxy-book-go/qcvss7180.mbn";
+};
+
+&wifi {
+	status = "okay";
+	vdd-0.8-cx-mx-supply = <&vreg_l9a_0p6>;
+	vdd-1.8-xo-supply = <&vreg_l1c_1p8>;
+	vdd-1.3-rfa-supply = <&vreg_l2c_1p3>;
+	vdd-3.3-ch0-supply = <&vreg_l10c_3p3>;
+	vdd-3.3-ch1-supply = <&vreg_l11c_3p3>;
 };
 
 &apps_rsc {
-	regulators-0 {
+	pm6150-rpmh-regulators {
 		compatible = "qcom,pm6150-rpmh-regulators";
 		qcom,pmic-id = "a";
 
-		vreg_s1a: smps1 {
+		vreg_s1a_1p1: smps1 {
 			regulator-min-microvolt = <1128000>;
 			regulator-max-microvolt = <1128000>;
 		};
 
-		vreg_l4a: ldo4 {
+		vreg_l4a_0p8: ldo4 { // acpi p88
 			regulator-min-microvolt = <824000>;
 			regulator-max-microvolt = <928000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l9a: ldo9 {
+		vreg_l9a_0p6: ldo9 { // acpi p664
 			regulator-min-microvolt = <488000>;
 			regulator-max-microvolt = <800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l10a: ldo10 {
+		vreg_l10a_1p8: ldo10 { // acpi 1p8 sx -> Vio
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
@@ -81,131 +682,131 @@ vreg_l10a: ldo10 {
 			regulator-boot-on;
 		};
 
-		vreg_l11a: ldo11 {
+		vreg_l11a_1p8: ldo11 { // trogdor
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l12a: ldo12 {
+		vreg_l12a_1p8: ldo12 { // acpi 1p8
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l13a: ldo13 {
+		vreg_l13a_1p8: ldo13 { // sx 1p8 NC
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l14a: ldo14 {
+		vreg_l14a_1p8: ldo14 { // sx 1p8 NC
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l15a: ldo15 {
+		vreg_l15a_1p8: ldo15 { // acpi 1p8 sx -> acodec
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l16a: ldo16 {
+		vreg_l16a_2p7: ldo16 { // sx 2p7 NC
 			regulator-min-microvolt = <2496000>;
 			regulator-max-microvolt = <3304000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l17a: ldo17 {
+		vreg_l17a_3p0: ldo17 { // acpi 3p088 sx NC
 			regulator-min-microvolt = <2920000>;
 			regulator-max-microvolt = <3232000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l18a: ldo18 {
+		vreg_l18a_2p8: ldo18 { // sx 3p3 NC
 			regulator-min-microvolt = <2496000>;
 			regulator-max-microvolt = <3304000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l19a: ldo19 {
+		vreg_l19a_2p9: ldo19 { // acpi 2p96
 			regulator-min-microvolt = <2960000>;
 			regulator-max-microvolt = <2960000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 	};
 
-	regulators-1 {
+	pm6150l-rpmh-regulators {
 		compatible = "qcom,pm6150l-rpmh-regulators";
 		qcom,pmic-id = "c";
 
-		vreg_s8c: smps8 {
+		vreg_s8c_1p3: smps8 { // sx 1p3 NC
 			regulator-min-microvolt = <1120000>;
 			regulator-max-microvolt = <1408000>;
 		};
 
-		vreg_l1c: ldo1 {
+		vreg_l1c_1p8: ldo1 { // acpi 1p8
 			regulator-min-microvolt = <1616000>;
 			regulator-max-microvolt = <1984000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l2c: ldo2 {
+		vreg_l2c_1p3: ldo2 { // acpi 1p304
 			regulator-min-microvolt = <1168000>;
 			regulator-max-microvolt = <1304000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l3c: ldo3 {
+		vreg_l3c_1p2: ldo3 { // acpi 1p2
 			regulator-min-microvolt = <1144000>;
 			regulator-max-microvolt = <1304000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l4c: ldo4 {
+		vreg_l4c_1p8: ldo4 { // sx 1p8 -> UIM1 (nc)
 			regulator-min-microvolt = <1648000>;
 			regulator-max-microvolt = <3304000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_LPM>;
 		};
 
-		vreg_l5c: ldo5 {
+		vreg_l5c_1p8: ldo5 { // sx 1p8 -> e-UIM (nc)
 			regulator-min-microvolt = <1648000>;
 			regulator-max-microvolt = <3304000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_LPM>;
 		};
 
-		vreg_l6c: ldo6 {
+		vreg_l6c_2p9: ldo6 { // sx 2p96 NC?
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <2950000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l7c: ldo7 {
+		vreg_l7c_3p0: ldo7 { // sx 3p0 NC?
 			regulator-min-microvolt = <3000000>;
 			regulator-max-microvolt = <3312000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_LPM>;
 		};
 
-		vreg_l8c: ldo8 {
+		vreg_l8c_1p8: ldo8 { // acpi 1p8 sx -> BRIJ
 			regulator-min-microvolt = <1800000>;
 			regulator-max-microvolt = <1800000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l9c: ldo9 {
+		vreg_l9c_2p9: ldo9 { // sx 2p95 NC
 			regulator-min-microvolt = <2952000>;
 			regulator-max-microvolt = <2952000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l10c: ldo10 {
+		vreg_l10c_3p3: ldo10 { // acpi 3p304
 			regulator-min-microvolt = <3000000>;
 			regulator-max-microvolt = <3400000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
 		};
 
-		vreg_l11c: ldo11 {
+		vreg_l11c_3p3: ldo11 {
 			regulator-min-microvolt = <3000000>;
 			regulator-max-microvolt = <3400000>;
 			regulator-initial-mode = <RPMH_REGULATOR_MODE_HPM>;
@@ -219,131 +820,397 @@ vreg_bob: bob {
 	};
 };
 
-&camcc {
-	status = "disabled";
+&qup_i2c0_default {
+	pinconf {
+// 0x22 and 0x23 for I2C1 in dsl - not defined for acer
+		pins = "gpio34", "gpio35";
+		drive-strength = <16>;
+
+		/* Has external pullup */
+		bias-disable;
+	};
 };
 
-&dispcc {
-	status = "disabled";
+&qup_i2c2_default {
+	pinconf {
+// 0x0F and 0x10 for I2C3 in dsl - same as for acer
+		pins = "gpio15", "gpio16";
+		drive-strength = <16>;
+
+		/* Has external pullup */
+		bias-disable;
+	};
 };
 
-&lpasscc {
-	status = "disabled";
+&qup_i2c4_default {
+	pinconf {
+// 0x73 and 0x74 for I2C5 in dsl - same as for acer
+		pins = "gpio115", "gpio116";
+		drive-strength = <16>;
+
+		/* Has external pullup */
+		bias-disable;
+	};
 };
 
-&lpass_hm {
-	status = "disabled";
+&qup_i2c5_default {
+	pinconf {
+// 0x19 and 0x1A for I2C6 in dsl - not defined for acer
+		pins = "gpio25", "gpio26";
+		drive-strength = <16>;
+
+		/* Has external pullup */
+		bias-disable;
+	};
 };
 
-&pm6150_pon {
-	status = "disabled";
+&qup_i2c6_default {
+	pinconf {
+// 0x3B and 0x3C for I2C7 in dsl - not defined for acer
+		pins = "gpio59", "gpio60";
+		drive-strength = <16>;
+
+		/* Has external pullup */
+		bias-disable;
+	};
 };
 
-&sdhc_2 {
-	pinctrl-names = "default", "sleep";
-	pinctrl-0 = <&sdc2_on>;
-	pinctrl-1 = <&sdc2_off>;
-	vmmc-supply = <&vreg_l9c>;
-	vqmmc-supply = <&vreg_l6c>;
-	cd-gpios = <&tlmm 69 GPIO_ACTIVE_LOW>;
-	status = "okay";
+&qup_i2c7_default {
+	pinconf {
+// 0x06 and 0x07 for I2C8 in dsl - not defined for acer
+		pins = "gpio6", "gpio7";
+		drive-strength = <16>;
+
+		/* Has external pullup */
+		bias-disable;
+	};
+};
+
+// OLD - no i2c9 on galaxy
+// &qup_i2c9_default {
+// 	pinconf {
+// 		pins = "gpio46", "gpio47";
+// 		drive-strength = <16>;
+// 
+// 		/* Has external pullup */
+// 		bias-disable;
+// 	};
+// };
+
+&qup_i2c10_default {
+	pinconf {
+// 0x56 and 0x57 for IC11 in dsl - was gpio86 and gpio87 also before, but there for the bridge which is i2c0 now with different gpios
+		pins = "gpio86", "gpio87";
+		drive-strength = <16>;
+
+		/* Has external pullup */
+		bias-disable;
+	};
 };
 
 &tlmm {
-	sdc2_on: sdc2-on-state {
-		clk-pins {
+// seems not really required on galaxy, so lets leave it off for now
+//	/*
+//	 * The TZ seem to protect those because some boards can have
+//	 * fingerprint sensor connected to this range. Not connected
+//	 * on this board
+//	 */
+//	gpio-reserved-ranges = <58 5>;
+//
+//	hid_keyboard_default: hid-keyboard-default {
+//// TODO - maybe gpio59 and/or gpio60 due to 0x3B and 0x3C for I2C7 in galaxy dsl
+//// UPDATE - this does not seem to work ...
+////		pins = "gpio33";
+//		pins = "gpio59";
+//		function = "gpio";
+//		drive-strength = <2>;
+//		bias-disable;
+//	};
+
+	bridge_en_default: bridge-en-default {
+// TODO
+		pins = "gpio51";
+		function = "gpio";
+		drive-strength = <16>;
+		bias-disable;
+	};
+
+	edp_bridge_irq_default: edp-bridge-irq-default {
+// TODO
+		pins = "gpio11";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-down;
+	};
+
+	bridge_suspend_default: bridge-suspend-default {
+// TODO
+		pins = "gpio22";
+		function = "gpio";
+		drive-strength = <16>;
+		bias-pull-up;
+	};
+
+// NEW - touchpad - based on aceridus work
+//  	en_touchpad: en-touchpad {
+//  		pinmux {
+//  			pins = "gpio31";
+//  			function = "gpio";
+//  		};
+//
+//  		pinconf {
+//  			pins = "gpio31";
+//  			drive-strength = <2>;
+//  			bias-disable;
+//  		};
+//  	};
+// ALTERNATIVE - based on old reg_tp_en_default below
+	en_touchpad: en_touchpad {
+// 		pins = "gpio31";
+// CHANGED - above value was used so far, but the galaxy dsl says 0x20 = 32 in TPXC
+		pins = "gpio32";
+// CHANGED - see LABEL_01 above
+//		pins = "gpio31";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+// OLD - touchpad
+//	hid_touchpad_default: hid-touchpad-default {
+//		pins = "gpio94";
+//		function = "gpio";
+//		drive-strength = <2>;
+//		bias-disable;
+//	};
+
+// IMPORTANT - all this P001 and P002 stuff below needs to be rechecked, maybe i got something wrong
+	reg_edp_1p2_en_default: reg-edp-1p2-en-deault {
+// TODO - not sure: gpio19 = 0x13 found as second TLMMGPIO for P002 in acer dsl which looks similar or P001 in galaxy dsl
+//        and the second TLMMGPIO there would be 0X1D, but the structure of P002/acer is a bit different from P001/galaxy
+//      - similarity found via "mdss_core_gdsc" string in the entry i found the acer gpio in and then searching for that
+//        entry of something similar in the galaxy dsl
+		pins = "gpio19";
+		function = "gpio";
+		drive-strength = <16>;
+		bias-disable;
+	};
+
+	reg_edp_1p8_en_default: reg-edp-1p8-en-deault {
+// TODO - not sure: gpio20 = 0x14 found as third TLMMGPIO for P002 in acer dsl which looks similar or P001 in galaxy dsl
+//        and the third TLMMGPIO there would be 0X1C, but the structure of P002/acer is a bit different from P001/galaxy
+		pins = "gpio20";
+		function = "gpio";
+		drive-strength = <16>;
+		bias-disable;
+	};
+
+	reg_lcm_en_default: reg-lcm-en-deault {
+// TODO - not sure: gpio26 = 0x1A found as first TLMMGPIO for P001 in acer dsl which looks similar or P001 in galaxy dsl
+//        and the first TLMMGPIO there would be 0X02 or 0x24, but the structure of P001/acer is a bit different from P001/galaxy
+//        and also P001/galaxy has already been used for the two entries above ...
+		pins = "gpio26";
+		function = "gpio";
+		drive-strength = <16>;
+		bias-disable;
+	};
+
+// LATER - this is audio - lets do this later ...
+// 	reg_audio_en_default: reg-audio-en-deault {
+// 		pins = "gpio83";
+// 		function = "gpio";
+// 		drive-strength = <2>;
+// 		bias-disable;
+// 	};
+
+// OLD - touchpad enable
+// 	reg_tp_en_default: reg-tp-en-deault {
+// 		pins = "gpio25";
+// 		function = "gpio";
+// 		drive-strength = <2>;
+// 		bias-disable;
+// 	};
+
+// LATER - this is audio - lets do this later ...
+// 	codec_irq_default: codec-irq-deault {
+// 		pins = "gpio28";
+// 		function = "gpio";
+// 		drive-strength = <2>;
+// 		bias-disable;
+// 	};
+
+	soc_bkoff_default: soc-bkoff-deault {
+// TODO
+		pins = "gpio10";
+		function = "gpio";
+		drive-strength = <16>;
+		bias-disable;
+	};
+
+	amp_sd_mode_default: amp-sd-mode-deault {
+// TODO - maybe audio too - so maybe lower prio?
+		pins = "gpio23";
+		function = "gpio";
+		drive-strength = <16>;
+		bias-disable;
+	};
+
+	ter_mi2s_active: ter-mi2s-active {
+// TODO
+		pins = "gpio63", "gpio64", "gpio65";
+		function = "mi2s_2";
+	};
+
+// OLD - we use sdc2 below as it worked for aceridus
+// 	sdc1_default: sdc1-default {
+// 		pinconf-clk {
+// 			pins = "sdc1_clk";
+// 			bias-disable;
+// 			drive-strength = <16>;
+// 		};
+// 
+// 		pinconf-cmd {
+// 			pins = "sdc1_cmd";
+// 			bias-pull-up;
+// 			drive-strength = <16>;
+// 		};
+// 
+// 		pinconf-data {
+// 			pins = "sdc1_data";
+// 			bias-pull-up;
+// 			drive-strength = <16>;
+// 		};
+// 
+// 		pinconf-rclk {
+// 			pins = "sdc1_rclk";
+// 			bias-pull-down;
+// 		};
+// 	};
+// 
+// 	sdc1_sleep: sdc1-sleep {
+// 		pinconf-clk {
+// 			pins = "sdc1_clk";
+// 			bias-disable;
+// 			drive-strength = <2>;
+// 		};
+// 
+// 		pinconf-cmd {
+// 			pins = "sdc1_cmd";
+// 			bias-pull-up;
+// 			drive-strength = <2>;
+// 		};
+// 
+// 		pinconf-data {
+// 			pins = "sdc1_data";
+// 			bias-pull-up;
+// 			drive-strength = <2>;
+// 		};
+// 
+// 		pinconf-rclk {
+// 			pins = "sdc1_rclk";
+// 			bias-pull-down;
+// 		};
+// 	};
+
+// NEW - lets use sdc2 as it worked for aceridus
+	sdc2_on: sdc2-on {
+		pinconf-clk {
 			pins = "sdc2_clk";
 			bias-disable;
 			drive-strength = <16>;
 		};
 
-		cmd-pins {
+		pinconf-cmd {
 			pins = "sdc2_cmd";
 			bias-pull-up;
 			drive-strength = <10>;
 		};
 
-		data-pins {
+		pinconf-data {
 			pins = "sdc2_data";
 			bias-pull-up;
 			drive-strength = <10>;
 		};
 
-		sd-cd-pins {
+		pinconf-sd-cd {
 			pins = "gpio69";
-			function = "gpio";
 			bias-pull-up;
 			drive-strength = <2>;
 		};
 	};
 
-	sdc2_off: sdc2-off-state {
-		clk-pins {
+	sdc2_off: sdc2-off {
+		pinconf-clk {
 			pins = "sdc2_clk";
 			bias-disable;
 			drive-strength = <2>;
 		};
 
-		cmd-pins {
+		pinconf-cmd {
 			pins = "sdc2_cmd";
 			bias-pull-up;
 			drive-strength = <2>;
 		};
 
-		data-pins {
+		pinconf-data {
 			pins = "sdc2_data";
 			bias-pull-up;
 			drive-strength = <2>;
 		};
 
-		sd-cd-pins {
+		pinconf-sd-cd {
 			pins = "gpio69";
-			function = "gpio";
 			bias-pull-up;
 			drive-strength = <2>;
 		};
 	};
-};
 
-&ufs_mem_hc {
-	vcc-supply = <&vreg_l19a>;
-	vcc-max-microamp = <600000>;
-	vccq2-supply = <&vreg_l12a>;
-	vccq2-max-microamp = <600000>;
-	status = "okay";
-};
-
-&ufs_mem_phy {
-	vdda-phy-supply = <&vreg_l4a>;
-	vdda-pll-supply = <&vreg_l3c>;
-	vdda-phy-max-microamp = <62900>;
-	vdda-pll-max-microamp = <18300>;
-	status = "okay";
-};
-
-&usb_1 {
-	status = "okay";
-};
 
-&usb_1_dwc3 {
-	dr_mode = "host";
-};
-
-&usb_1_hsphy {
-	vdd-supply = <&vreg_l4a>;
-	vdda-pll-supply = <&vreg_l11a>;
-	vdda-phy-dpdm-supply = <&vreg_l17a>;
-	qcom,imp-res-offset-value = <8>;
-	qcom,preemphasis-level = <QUSB2_V2_PREEMPHASIS_15_PERCENT>;
-	qcom,preemphasis-width = <QUSB2_V2_PREEMPHASIS_WIDTH_HALF_BIT>;
-	qcom,bias-ctrl-value = <0x22>;
-	qcom,charge-ctrl-value = <3>;
-	qcom,hsdisc-trim-value = <0>;
-	status = "okay";
-};
-
-&usb_1_qmpphy {
-	vdda-phy-supply = <&vreg_l3c>;
-	vdda-pll-supply = <&vreg_l4a>;
-	status = "okay";
+// LATER - this is bluetooth - lets do this later ...
+// 	qup_uart3_sleep: qup-uart3-sleep {
+// 		pinmux {
+// 			pins = "gpio38", "gpio39",
+// 			       "gpio40", "gpio41";
+// 			function = "gpio";
+// 		};
+// 
+// 		pinconf-cts {
+// 			/*
+// 			 * Configure a pull-down on CTS to match the pull of
+// 			 * the Bluetooth module.
+// 			 */
+// 			pins = "gpio38";
+// 			bias-pull-down;
+// 		};
+// 
+// 		pinconf-rts {
+// 			/*
+// 			 * Configure pull-down on RTS. As RTS is active low
+// 			 * signal, pull it low to indicate the BT SoC that it
+// 			 * can wakeup the system anytime from suspend state by
+// 			 * pulling RX low (by sending wakeup bytes).
+// 			 */
+// 			 pins = "gpio39";
+// 			 bias-pull-down;
+// 		};
+// 
+// 		pinconf-tx {
+// 			/*
+// 			 * Configure pull-up on TX when it isn't actively driven
+// 			 * to prevent BT SoC from receiving garbage during sleep.
+// 			 */
+// 			pins = "gpio40";
+// 			bias-pull-up;
+// 		};
+// 
+// 		pinconf-rx {
+// 			/*
+// 			 * Configure a pull-up on RX. This is needed to avoid
+// 			 * garbage data when the TX pin of the Bluetooth module
+// 			 * is floating which may cause spurious wakeups.
+// 			 */
+// 			pins = "gpio41";
+// 			bias-pull-up;
+// 		};
+// 	};
 };
diff --git a/arch/arm64/boot/dts/qcom/sc7180.dtsi b/arch/arm64/boot/dts/qcom/sc7180.dtsi
index 95d334e2e..6b59a592f 100644
--- a/arch/arm64/boot/dts/qcom/sc7180.dtsi
+++ b/arch/arm64/boot/dts/qcom/sc7180.dtsi
@@ -10,7 +10,10 @@
 #include <dt-bindings/clock/qcom,gpucc-sc7180.h>
 #include <dt-bindings/clock/qcom,lpasscorecc-sc7180.h>
 #include <dt-bindings/clock/qcom,rpmh.h>
+#include <dt-bindings/clock/qcom,sm8250-lpass-aoncc.h>
+#include <dt-bindings/clock/qcom,sm8250-lpass-audiocc.h>
 #include <dt-bindings/clock/qcom,videocc-sc7180.h>
+#include <dt-bindings/interconnect/qcom,icc.h>
 #include <dt-bindings/interconnect/qcom,osm-l3.h>
 #include <dt-bindings/interconnect/qcom,sc7180.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
@@ -19,6 +22,8 @@
 #include <dt-bindings/reset/qcom,sdm845-aoss.h>
 #include <dt-bindings/reset/qcom,sdm845-pdc.h>
 #include <dt-bindings/soc/qcom,rpmh-rsc.h>
+#include <dt-bindings/soc/qcom,apr.h>
+#include <dt-bindings/sound/qcom,q6afe.h>
 #include <dt-bindings/thermal/thermal.h>
 
 / {
@@ -78,14 +83,12 @@ CPU0: cpu@0 {
 			reg = <0x0 0x0>;
 			clocks = <&cpufreq_hw 0>;
 			enable-method = "psci";
-			cpu-idle-states = <&LITTLE_CPU_SLEEP_0
-					   &LITTLE_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD0>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <415>;
 			dynamic-power-coefficient = <137>;
 			operating-points-v2 = <&cpu0_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			next-level-cache = <&L2_0>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 0>;
@@ -108,15 +111,13 @@ CPU1: cpu@100 {
 			reg = <0x0 0x100>;
 			clocks = <&cpufreq_hw 0>;
 			enable-method = "psci";
-			cpu-idle-states = <&LITTLE_CPU_SLEEP_0
-					   &LITTLE_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD1>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <415>;
 			dynamic-power-coefficient = <137>;
 			next-level-cache = <&L2_100>;
 			operating-points-v2 = <&cpu0_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 0>;
 			L2_100: l2-cache {
@@ -133,15 +134,13 @@ CPU2: cpu@200 {
 			reg = <0x0 0x200>;
 			clocks = <&cpufreq_hw 0>;
 			enable-method = "psci";
-			cpu-idle-states = <&LITTLE_CPU_SLEEP_0
-					   &LITTLE_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD2>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <415>;
 			dynamic-power-coefficient = <137>;
 			next-level-cache = <&L2_200>;
 			operating-points-v2 = <&cpu0_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 0>;
 			L2_200: l2-cache {
@@ -158,15 +157,13 @@ CPU3: cpu@300 {
 			reg = <0x0 0x300>;
 			clocks = <&cpufreq_hw 0>;
 			enable-method = "psci";
-			cpu-idle-states = <&LITTLE_CPU_SLEEP_0
-					   &LITTLE_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD3>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <415>;
 			dynamic-power-coefficient = <137>;
 			next-level-cache = <&L2_300>;
 			operating-points-v2 = <&cpu0_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 0>;
 			L2_300: l2-cache {
@@ -183,15 +180,13 @@ CPU4: cpu@400 {
 			reg = <0x0 0x400>;
 			clocks = <&cpufreq_hw 0>;
 			enable-method = "psci";
-			cpu-idle-states = <&LITTLE_CPU_SLEEP_0
-					   &LITTLE_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD4>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <415>;
 			dynamic-power-coefficient = <137>;
 			next-level-cache = <&L2_400>;
 			operating-points-v2 = <&cpu0_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 0>;
 			L2_400: l2-cache {
@@ -208,15 +203,13 @@ CPU5: cpu@500 {
 			reg = <0x0 0x500>;
 			clocks = <&cpufreq_hw 0>;
 			enable-method = "psci";
-			cpu-idle-states = <&LITTLE_CPU_SLEEP_0
-					   &LITTLE_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD5>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <415>;
 			dynamic-power-coefficient = <137>;
 			next-level-cache = <&L2_500>;
 			operating-points-v2 = <&cpu0_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 0>;
 			L2_500: l2-cache {
@@ -233,15 +226,13 @@ CPU6: cpu@600 {
 			reg = <0x0 0x600>;
 			clocks = <&cpufreq_hw 1>;
 			enable-method = "psci";
-			cpu-idle-states = <&BIG_CPU_SLEEP_0
-					   &BIG_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD6>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <1024>;
 			dynamic-power-coefficient = <480>;
 			next-level-cache = <&L2_600>;
 			operating-points-v2 = <&cpu6_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 1>;
 			L2_600: l2-cache {
@@ -258,15 +249,13 @@ CPU7: cpu@700 {
 			reg = <0x0 0x700>;
 			clocks = <&cpufreq_hw 1>;
 			enable-method = "psci";
-			cpu-idle-states = <&BIG_CPU_SLEEP_0
-					   &BIG_CPU_SLEEP_1
-					   &CLUSTER_SLEEP_0>;
+			power-domains = <&CPU_PD7>;
+			power-domain-names = "psci";
 			capacity-dmips-mhz = <1024>;
 			dynamic-power-coefficient = <480>;
 			next-level-cache = <&L2_700>;
 			operating-points-v2 = <&cpu6_opp_table>;
-			interconnects = <&gem_noc MASTER_APPSS_PROC 3 &mc_virt SLAVE_EBI1 3>,
-					<&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
+			interconnects = <&osm_l3 MASTER_OSM_L3_APPS &osm_l3 SLAVE_OSM_L3>;
 			#cooling-cells = <2>;
 			qcom,freq-domain = <&cpufreq_hw 1>;
 			L2_700: l2-cache {
@@ -355,15 +344,34 @@ BIG_CPU_SLEEP_1: cpu-sleep-1-1 {
 				min-residency-us = <5555>;
 				local-timer-stop;
 			};
+		};
 
-			CLUSTER_SLEEP_0: cluster-sleep-0 {
-				compatible = "arm,idle-state";
+		domain-idle-states {
+			CLUSTER_SLEEP_PC: cluster-sleep-0 {
+				compatible = "domain-idle-state";
+				idle-state-name = "cluster-l3-power-collapse";
+				arm,psci-suspend-param = <0x41000044>;
+				entry-latency-us = <2752>;
+				exit-latency-us = <3048>;
+				min-residency-us = <6118>;
+			};
+
+			CLUSTER_SLEEP_CX_RET: cluster-sleep-1 {
+				compatible = "domain-idle-state";
+				idle-state-name = "cluster-cx-retention";
+				arm,psci-suspend-param = <0x41001244>;
+				entry-latency-us = <3638>;
+				exit-latency-us = <4562>;
+				min-residency-us = <8467>;
+			};
+
+			CLUSTER_AOSS_SLEEP: cluster-sleep-2 {
+				compatible = "domain-idle-state";
 				idle-state-name = "cluster-power-down";
-				arm,psci-suspend-param = <0x40003444>;
+				arm,psci-suspend-param = <0x4100b244>;
 				entry-latency-us = <3263>;
 				exit-latency-us = <6562>;
-				min-residency-us = <9926>;
-				local-timer-stop;
+				min-residency-us = <9826>;
 			};
 		};
 	};
@@ -386,52 +394,52 @@ cpu0_opp_table: opp-table-cpu0 {
 
 		cpu0_opp1: opp-300000000 {
 			opp-hz = /bits/ 64 <300000000>;
-			opp-peak-kBps = <1200000 4800000>;
+			opp-peak-kBps = <4800000>;
 		};
 
 		cpu0_opp2: opp-576000000 {
 			opp-hz = /bits/ 64 <576000000>;
-			opp-peak-kBps = <1200000 4800000>;
+			opp-peak-kBps = <4800000>;
 		};
 
 		cpu0_opp3: opp-768000000 {
 			opp-hz = /bits/ 64 <768000000>;
-			opp-peak-kBps = <1200000 4800000>;
+			opp-peak-kBps = <4800000>;
 		};
 
 		cpu0_opp4: opp-1017600000 {
 			opp-hz = /bits/ 64 <1017600000>;
-			opp-peak-kBps = <1804000 8908800>;
+			opp-peak-kBps = <8908800>;
 		};
 
 		cpu0_opp5: opp-1248000000 {
 			opp-hz = /bits/ 64 <1248000000>;
-			opp-peak-kBps = <2188000 12902400>;
+			opp-peak-kBps = <12902400>;
 		};
 
 		cpu0_opp6: opp-1324800000 {
 			opp-hz = /bits/ 64 <1324800000>;
-			opp-peak-kBps = <2188000 12902400>;
+			opp-peak-kBps = <12902400>;
 		};
 
 		cpu0_opp7: opp-1516800000 {
 			opp-hz = /bits/ 64 <1516800000>;
-			opp-peak-kBps = <3072000 15052800>;
+			opp-peak-kBps = <15052800>;
 		};
 
 		cpu0_opp8: opp-1612800000 {
 			opp-hz = /bits/ 64 <1612800000>;
-			opp-peak-kBps = <3072000 15052800>;
+			opp-peak-kBps = <15052800>;
 		};
 
 		cpu0_opp9: opp-1708800000 {
 			opp-hz = /bits/ 64 <1708800000>;
-			opp-peak-kBps = <3072000 15052800>;
+			opp-peak-kBps = <15052800>;
 		};
 
 		cpu0_opp10: opp-1804800000 {
 			opp-hz = /bits/ 64 <1804800000>;
-			opp-peak-kBps = <4068000 22425600>;
+			opp-peak-kBps = <22425600>;
 		};
 	};
 
@@ -441,82 +449,82 @@ cpu6_opp_table: opp-table-cpu6 {
 
 		cpu6_opp1: opp-300000000 {
 			opp-hz = /bits/ 64 <300000000>;
-			opp-peak-kBps = <2188000 8908800>;
+			opp-peak-kBps = <8908800>;
 		};
 
 		cpu6_opp2: opp-652800000 {
 			opp-hz = /bits/ 64 <652800000>;
-			opp-peak-kBps = <2188000 8908800>;
+			opp-peak-kBps = <8908800>;
 		};
 
 		cpu6_opp3: opp-825600000 {
 			opp-hz = /bits/ 64 <825600000>;
-			opp-peak-kBps = <2188000 8908800>;
+			opp-peak-kBps = <8908800>;
 		};
 
 		cpu6_opp4: opp-979200000 {
 			opp-hz = /bits/ 64 <979200000>;
-			opp-peak-kBps = <2188000 8908800>;
+			opp-peak-kBps = <8908800>;
 		};
 
 		cpu6_opp5: opp-1113600000 {
 			opp-hz = /bits/ 64 <1113600000>;
-			opp-peak-kBps = <2188000 8908800>;
+			opp-peak-kBps = <8908800>;
 		};
 
 		cpu6_opp6: opp-1267200000 {
 			opp-hz = /bits/ 64 <1267200000>;
-			opp-peak-kBps = <4068000 12902400>;
+			opp-peak-kBps = <12902400>;
 		};
 
 		cpu6_opp7: opp-1555200000 {
 			opp-hz = /bits/ 64 <1555200000>;
-			opp-peak-kBps = <4068000 15052800>;
+			opp-peak-kBps = <15052800>;
 		};
 
 		cpu6_opp8: opp-1708800000 {
 			opp-hz = /bits/ 64 <1708800000>;
-			opp-peak-kBps = <6220000 19353600>;
+			opp-peak-kBps = <19353600>;
 		};
 
 		cpu6_opp9: opp-1843200000 {
 			opp-hz = /bits/ 64 <1843200000>;
-			opp-peak-kBps = <6220000 19353600>;
+			opp-peak-kBps = <19353600>;
 		};
 
 		cpu6_opp10: opp-1900800000 {
 			opp-hz = /bits/ 64 <1900800000>;
-			opp-peak-kBps = <6220000 22425600>;
+			opp-peak-kBps = <22425600>;
 		};
 
 		cpu6_opp11: opp-1996800000 {
 			opp-hz = /bits/ 64 <1996800000>;
-			opp-peak-kBps = <6220000 22425600>;
+			opp-peak-kBps = <22425600>;
 		};
 
 		cpu6_opp12: opp-2112000000 {
 			opp-hz = /bits/ 64 <2112000000>;
-			opp-peak-kBps = <6220000 22425600>;
+			opp-peak-kBps = <22425600>;
 		};
 
 		cpu6_opp13: opp-2208000000 {
 			opp-hz = /bits/ 64 <2208000000>;
-			opp-peak-kBps = <7216000 22425600>;
+			opp-peak-kBps = <22425600>;
 		};
 
 		cpu6_opp14: opp-2323200000 {
 			opp-hz = /bits/ 64 <2323200000>;
-			opp-peak-kBps = <7216000 22425600>;
+			opp-peak-kBps = <22425600>;
 		};
 
 		cpu6_opp15: opp-2400000000 {
 			opp-hz = /bits/ 64 <2400000000>;
-			opp-peak-kBps = <8532000 23347200>;
+			opp-peak-kBps = <23347200>;
 		};
 
 		cpu6_opp16: opp-2553600000 {
 			opp-hz = /bits/ 64 <2553600000>;
-			opp-peak-kBps = <8532000 23347200>;
+			opp-peak-kBps = <23347200>;
 		};
 	};
 
@@ -566,6 +574,61 @@ pmu {
 	psci {
 		compatible = "arm,psci-1.0";
 		method = "smc";
+
+		CPU_PD0: cpu0 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&LITTLE_CPU_SLEEP_0 &LITTLE_CPU_SLEEP_1>;
+		};
+
+		CPU_PD1: cpu1 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&LITTLE_CPU_SLEEP_0 &LITTLE_CPU_SLEEP_1>;
+		};
+
+		CPU_PD2: cpu2 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&LITTLE_CPU_SLEEP_0 &LITTLE_CPU_SLEEP_1>;
+		};
+
+		CPU_PD3: cpu3 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&LITTLE_CPU_SLEEP_0 &LITTLE_CPU_SLEEP_1>;
+		};
+
+		CPU_PD4: cpu4 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&LITTLE_CPU_SLEEP_0 &LITTLE_CPU_SLEEP_1>;
+		};
+
+		CPU_PD5: cpu5 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&LITTLE_CPU_SLEEP_0 &LITTLE_CPU_SLEEP_1>;
+		};
+
+		CPU_PD6: cpu6 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&BIG_CPU_SLEEP_0 &BIG_CPU_SLEEP_1>;
+		};
+
+		CPU_PD7: cpu7 {
+			#power-domain-cells = <0>;
+			power-domains = <&CLUSTER_PD>;
+			domain-idle-states = <&BIG_CPU_SLEEP_0 &BIG_CPU_SLEEP_1>;
+		};
+
+		CLUSTER_PD: cpu-cluster0 {
+			#power-domain-cells = <0>;
+			domain-idle-states = <&CLUSTER_SLEEP_PC
+					      &CLUSTER_SLEEP_CX_RET
+					      &CLUSTER_AOSS_SLEEP>;
+		};
 	};
 
 	reserved_memory: reserved-memory {
@@ -2722,6 +2785,154 @@ apss_merge_funnel_in: endpoint {
 			};
 		};
 
+		remoteproc_adsp: remoteproc@62400000 {
+			compatible = "qcom,sc7180-adsp-pas";
+			reg = <0 0x62400000 0 0x100>;
+
+			interrupts-extended = <&intc GIC_SPI 162 IRQ_TYPE_EDGE_RISING>,
+						<&adsp_smp2p_in 0 IRQ_TYPE_EDGE_RISING>,
+						<&adsp_smp2p_in 1 IRQ_TYPE_EDGE_RISING>,
+						<&adsp_smp2p_in 2 IRQ_TYPE_EDGE_RISING>,
+						<&adsp_smp2p_in 3 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "wdog", "fatal", "ready",
+						"handover", "stop-ack";
+
+			clocks = <&rpmhcc RPMH_CXO_CLK>;
+			clock-names = "xo";
+
+			power-domains = <&rpmhpd SC7180_LCX>,
+					<&rpmhpd SC7180_LMX>;
+			power-domain-names = "lcx", "lmx";
+
+			qcom,qmp = <&aoss_qmp>;
+
+			qcom,smem-states = <&adsp_smp2p_out 0>;
+			qcom,smem-state-names = "stop";
+
+			status = "disabled";
+
+			glink-edge {
+				interrupts = <GIC_SPI 156 IRQ_TYPE_EDGE_RISING>;
+				label = "lpass";
+				qcom,remote-pid = <2>;
+				mboxes = <&apss_shared 8>;
+
+				apr {
+					compatible = "qcom,apr-v2";
+					qcom,glink-channels = "apr_audio_svc";
+					qcom,apr-domain = <APR_DOMAIN_ADSP>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					apr-service@3 {
+						reg = <APR_SVC_ADSP_CORE>;
+						compatible = "qcom,q6core";
+						qcom,protection-domain = "avs/audio", "msm/adsp/audio_pd";
+					};
+
+					q6afe: apr-service@4 {
+						compatible = "qcom,q6afe";
+						reg = <APR_SVC_AFE>;
+						qcom,protection-domain = "avs/audio", "msm/adsp/audio_pd";
+						q6afedai: dais {
+							compatible = "qcom,q6afe-dais";
+							#address-cells = <1>;
+							#size-cells = <0>;
+							#sound-dai-cells = <1>;
+						};
+
+						q6afecc: cc {
+							compatible = "qcom,q6afe-clocks";
+							#clock-cells = <2>;
+						};
+					};
+
+					q6asm: apr-service@7 {
+						compatible = "qcom,q6asm";
+						reg = <APR_SVC_ASM>;
+						qcom,protection-domain = "avs/audio", "msm/adsp/audio_pd";
+						q6asmdai: dais {
+							compatible = "qcom,q6asm-dais";
+							#address-cells = <1>;
+							#size-cells = <0>;
+							#sound-dai-cells = <1>;
+							iommus = <&apps_smmu 0x1001 0x0>;
+						};
+					};
+
+					q6adm: apr-service@8 {
+						compatible = "qcom,q6adm";
+						reg = <APR_SVC_ADM>;
+						qcom,protection-domain = "avs/audio", "msm/adsp/audio_pd";
+						q6routing: routing {
+							compatible = "qcom,q6adm-routing";
+							#sound-dai-cells = <0>;
+						};
+					};
+				};
+
+				fastrpc {
+					compatible = "qcom,fastrpc";
+					qcom,glink-channels = "fastrpcglink-apps-dsp";
+					label = "adsp";
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					compute-cb@3 {
+						compatible = "qcom,fastrpc-compute-cb";
+						reg = <3>;
+						iommus = <&apps_smmu 0x1003 0x0>;
+					};
+
+					compute-cb@4 {
+						compatible = "qcom,fastrpc-compute-cb";
+						reg = <4>;
+						iommus = <&apps_smmu 0x1004 0x0>;
+					};
+
+					compute-cb@5 {
+						compatible = "qcom,fastrpc-compute-cb";
+						reg = <5>;
+						iommus = <&apps_smmu 0x1005 0x0>;
+						qcom,nsessions = <5>;
+					};
+				};
+			};
+		};
+
+		audiocc: clock-controller@62700000 {
+			compatible = "qcom,sm8250-lpass-audiocc";
+			reg = <0 0x62700000 0 0x30000>;
+			#clock-cells = <1>;
+			clocks = <&q6afecc LPASS_HW_MACRO_VOTE LPASS_CLK_ATTRIBUTE_COUPLE_NO>,
+				<&q6afecc LPASS_HW_DCODEC_VOTE LPASS_CLK_ATTRIBUTE_COUPLE_NO>,
+				<&q6afecc LPASS_CLK_ID_TX_CORE_MCLK LPASS_CLK_ATTRIBUTE_COUPLE_NO>;
+			clock-names = "core", "audio", "bus";
+		};
+
+		aoncc: clock-controller@62780000 {
+			compatible = "qcom,sm8250-lpass-aoncc";
+			reg = <0 0x62780000 0 0x40000>;
+			#clock-cells = <1>;
+			clocks = <&q6afecc LPASS_HW_MACRO_VOTE LPASS_CLK_ATTRIBUTE_COUPLE_NO>,
+				<&q6afecc LPASS_HW_DCODEC_VOTE LPASS_CLK_ATTRIBUTE_COUPLE_NO>,
+				<&q6afecc LPASS_CLK_ID_TX_CORE_NPL_MCLK LPASS_CLK_ATTRIBUTE_COUPLE_NO>;
+			clock-names = "core", "audio", "bus";
+		};
+
+		lpass_tlmm: pinctrl@627c0000 {
+			compatible = "qcom,sc7280-lpass-lpi-pinctrl";
+			reg = <0 0x627c0000 0x0 0x20000>,
+				<0 0x62950000 0x0 0x10000>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			gpio-ranges = <&lpass_tlmm 0 0 15>;
+
+			clocks = <&q6afecc LPASS_HW_MACRO_VOTE LPASS_CLK_ATTRIBUTE_COUPLE_NO>,
+				<&q6afecc LPASS_HW_DCODEC_VOTE LPASS_CLK_ATTRIBUTE_COUPLE_NO>;
+			clock-names = "core", "audio";
+		};
+
 		sdhc_2: mmc@8804000 {
 			compatible = "qcom,sc7180-sdhci", "qcom,sdhci-msm-v5";
 			reg = <0 0x08804000 0 0x1000>;
@@ -2841,6 +3052,90 @@ dp_phy: dp-phy@88ea200 {
 			};
 		};
 
+		pmu@90b6300 {
+			compatible = "qcom,sc7180-cpu-bwmon", "qcom,sdm845-bwmon";
+			reg = <0 0x090b6300 0 0x600>;
+			interrupts = <GIC_SPI 581 IRQ_TYPE_LEVEL_HIGH>;
+
+			interconnects = <&gem_noc MASTER_APPSS_PROC QCOM_ICC_TAG_ACTIVE_ONLY
+					 &gem_noc SLAVE_LLCC QCOM_ICC_TAG_ACTIVE_ONLY>;
+			operating-points-v2 = <&cpu_bwmon_opp_table>;
+
+			cpu_bwmon_opp_table: opp-table {
+				compatible = "operating-points-v2";
+
+				opp-0 {
+					opp-peak-kBps = <2288000>;
+				};
+
+				opp-1 {
+					opp-peak-kBps = <4577000>;
+				};
+
+				opp-2 {
+					opp-peak-kBps = <7110000>;
+				};
+
+				opp-3 {
+					opp-peak-kBps = <9155000>;
+				};
+
+				opp-4 {
+					opp-peak-kBps = <12298000>;
+				};
+
+				opp-5 {
+					opp-peak-kBps = <14236000>;
+				};
+			};
+		};
+
+		pmu@90cd000 {
+			compatible = "qcom,sc7180-llcc-bwmon", "qcom,sc7280-llcc-bwmon";
+			reg = <0 0x090cd000 0 0x1000>;
+			interrupts = <GIC_SPI 241 IRQ_TYPE_LEVEL_HIGH>;
+
+			interconnects = <&mc_virt MASTER_LLCC QCOM_ICC_TAG_ACTIVE_ONLY
+					 &mc_virt SLAVE_EBI1 QCOM_ICC_TAG_ACTIVE_ONLY>;
+			operating-points-v2 = <&llcc_bwmon_opp_table>;
+
+			llcc_bwmon_opp_table: opp-table {
+				compatible = "operating-points-v2";
+
+				opp-0 {
+					opp-peak-kBps = <1144000>;
+				};
+
+				opp-1 {
+					opp-peak-kBps = <1720000>;
+				};
+
+				opp-2 {
+					opp-peak-kBps = <2086000>;
+				};
+
+				opp-3 {
+					opp-peak-kBps = <2929000>;
+				};
+
+				opp-4 {
+					opp-peak-kBps = <3879000>;
+				};
+
+				opp-5 {
+					opp-peak-kBps = <5931000>;
+				};
+
+				opp-6 {
+					opp-peak-kBps = <6881000>;
+				};
+
+				opp-7 {
+					opp-peak-kBps = <8137000>;
+				};
+			};
+		};
+
 		dc_noc: interconnect@9160000 {
 			compatible = "qcom,sc7180-dc-noc";
 			reg = <0 0x09160000 0 0x03200>;
@@ -3143,8 +3438,6 @@ mdss_dsi0: dsi@ae94000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
-				status = "disabled";
-
 				ports {
 					#address-cells = <1>;
 					#size-cells = <0>;
@@ -3580,6 +3873,7 @@ apps_rsc: rsc@18200000 {
 					  <SLEEP_TCS   3>,
 					  <WAKE_TCS    3>,
 					  <CONTROL_TCS 1>;
+			power-domains = <&CLUSTER_PD>;
 
 			rpmhcc: clock-controller {
 				compatible = "qcom,sc7180-rpmh-clk";
diff --git a/drivers/gpu/drm/msm/disp/dpu1/dpu_kms.c b/drivers/gpu/drm/msm/disp/dpu1/dpu_kms.c
index aa8499de1..d652d1265 100644
--- a/drivers/gpu/drm/msm/disp/dpu1/dpu_kms.c
+++ b/drivers/gpu/drm/msm/disp/dpu1/dpu_kms.c
@@ -731,6 +731,8 @@ static int _dpu_kms_drm_obj_init(struct dpu_kms *dpu_kms)
 	struct drm_crtc *crtc;
 	struct drm_encoder *encoder;
 	unsigned int num_encoders;
+	unsigned primary_idx = 0;
+	unsigned cursor_idx = 0;
 
 	struct msm_drm_private *priv;
 	const struct dpu_mdss_cfg *catalog;
@@ -758,21 +760,28 @@ static int _dpu_kms_drm_obj_init(struct dpu_kms *dpu_kms)
 	/* Create the planes, keeping track of one primary/cursor per crtc */
 	for (i = 0; i < catalog->sspp_count; i++) {
 		enum drm_plane_type type;
+		unsigned possible_crtcs;
 
 		if ((catalog->sspp[i].features & BIT(DPU_SSPP_CURSOR))
-			&& cursor_planes_idx < max_crtc_count)
+			&& cursor_planes_idx < max_crtc_count) {
 			type = DRM_PLANE_TYPE_CURSOR;
-		else if (primary_planes_idx < max_crtc_count)
+			possible_crtcs = BIT(cursor_idx);
+			cursor_idx++;
+		} else if (primary_planes_idx < max_crtc_count) {
 			type = DRM_PLANE_TYPE_PRIMARY;
-		else
+			possible_crtcs = BIT(primary_idx);
+			primary_idx++;
+		} else {
 			type = DRM_PLANE_TYPE_OVERLAY;
+			possible_crtcs = (1UL << max_crtc_count) - 1;
+		}
 
 		DPU_DEBUG("Create plane type %d with features %lx (cur %lx)\n",
 			  type, catalog->sspp[i].features,
 			  catalog->sspp[i].features & BIT(DPU_SSPP_CURSOR));
 
 		plane = dpu_plane_init(dev, catalog->sspp[i].id, type,
-				       (1UL << max_crtc_count) - 1);
+				       possible_crtcs);
 		if (IS_ERR(plane)) {
 			DPU_ERROR("dpu_plane_init failed\n");
 			ret = PTR_ERR(plane);
diff --git a/drivers/gpu/drm/msm/disp/dpu1/dpu_plane.c b/drivers/gpu/drm/msm/disp/dpu1/dpu_plane.c
index c2aaaded0..9be8f170b 100644
--- a/drivers/gpu/drm/msm/disp/dpu1/dpu_plane.c
+++ b/drivers/gpu/drm/msm/disp/dpu1/dpu_plane.c
@@ -1416,7 +1416,7 @@ struct drm_plane *dpu_plane_init(struct drm_device *dev,
 	format_list = pipe_hw->cap->sblk->format_list;
 	num_formats = pipe_hw->cap->sblk->num_formats;
 
-	ret = drm_universal_plane_init(dev, plane, 0xff, &dpu_plane_funcs,
+	ret = drm_universal_plane_init(dev, plane, possible_crtcs, &dpu_plane_funcs,
 				format_list, num_formats,
 				supported_format_modifiers, type, NULL);
 	if (ret)
diff --git a/drivers/gpu/drm/panel/panel-edp.c b/drivers/gpu/drm/panel/panel-edp.c
index fbd114b4f..29121445a 100644
--- a/drivers/gpu/drm/panel/panel-edp.c
+++ b/drivers/gpu/drm/panel/panel-edp.c
@@ -1868,6 +1868,7 @@ static const struct edp_panel_entry edp_panels[] = {
 	EDP_PANEL_ENTRY('A', 'U', 'O', 0x1062, &delay_200_500_e50, "B120XAN01.0"),
 	EDP_PANEL_ENTRY('A', 'U', 'O', 0x1e9b, &delay_200_500_e50, "B133UAN02.1"),
 	EDP_PANEL_ENTRY('A', 'U', 'O', 0x1ea5, &delay_200_500_e50, "B116XAK01.6"),
+	EDP_PANEL_ENTRY('A', 'U', 'O', 0x203d, &delay_200_500_e50, "B140HTN02.0"),
 	EDP_PANEL_ENTRY('A', 'U', 'O', 0x405c, &auo_b116xak01.delay, "B116XAK01"),
 	EDP_PANEL_ENTRY('A', 'U', 'O', 0x582d, &delay_200_500_e50, "B133UAN01.0"),
 	EDP_PANEL_ENTRY('A', 'U', 'O', 0x615c, &delay_200_500_e50, "B116XAN06.1"),
@@ -1889,6 +1890,7 @@ static const struct edp_panel_entry edp_panels[] = {
 	EDP_PANEL_ENTRY('C', 'M', 'N', 0x1153, &delay_200_500_e80_d50, "N116BGE-EA2"),
 	EDP_PANEL_ENTRY('C', 'M', 'N', 0x1154, &delay_200_500_e80_d50, "N116BCA-EA2"),
 	EDP_PANEL_ENTRY('C', 'M', 'N', 0x1247, &delay_200_500_e80_d50, "N120ACA-EA1"),
+	EDP_PANEL_ENTRY('C', 'M', 'N', 0x14d4, &delay_200_500_e80_d50, "N140HCA-EAC"),
 
 	EDP_PANEL_ENTRY('I', 'V', 'O', 0x057d, &delay_200_500_e200, "R140NWF5 RH"),
 	EDP_PANEL_ENTRY('I', 'V', 'O', 0x854a, &delay_200_500_p2e100, "M133NW4J"),
diff --git a/drivers/media/platform/qcom/venus/core.c b/drivers/media/platform/qcom/venus/core.c
index 2ae867cb4..bee1b4c98 100644
--- a/drivers/media/platform/qcom/venus/core.c
+++ b/drivers/media/platform/qcom/venus/core.c
@@ -760,6 +760,10 @@ static const struct venus_resources sc7180_res = {
 	.vmem_size = 0,
 	.vmem_addr = 0,
 	.dma_mask = 0xe0000000 - 1,
+	.cp_start = 0,
+	.cp_size = 0x70800000,
+	.cp_nonpixel_start = 0x1000000,
+	.cp_nonpixel_size = 0x24800000,
 	.fwname = "qcom/venus-5.4/venus.mdt",
 };
 
diff --git a/drivers/power/supply/Kconfig b/drivers/power/supply/Kconfig
index 663a1c423..388ce37dd 100644
--- a/drivers/power/supply/Kconfig
+++ b/drivers/power/supply/Kconfig
@@ -951,4 +951,11 @@ config CHARGER_QCOM_SMB2
 	  adds support for the SMB2 switch mode battery charger found
 	  in PMI8998 and related PMICs.
 
+config ACER_ASPIRE_BATTERY
+	tristate "Aspire1 Battery Fuel Gauge"
+	depends on I2C
+	depends on DRM
+	help
+	  Of course!
+
 endif # POWER_SUPPLY
diff --git a/drivers/power/supply/Makefile b/drivers/power/supply/Makefile
index a8a9fa6de..8a292b810 100644
--- a/drivers/power/supply/Makefile
+++ b/drivers/power/supply/Makefile
@@ -110,4 +110,5 @@ obj-$(CONFIG_BATTERY_ACER_A500)	+= acer_a500_battery.o
 obj-$(CONFIG_BATTERY_SURFACE)	+= surface_battery.o
 obj-$(CONFIG_CHARGER_SURFACE)	+= surface_charger.o
 obj-$(CONFIG_BATTERY_UG3105)	+= ug3105_battery.o
+obj-$(CONFIG_ACER_ASPIRE_BATTERY)	+= aspire-ec-battery.o
 obj-$(CONFIG_CHARGER_QCOM_SMB2)	+= qcom_pmi8998_charger.o
diff --git a/drivers/remoteproc/qcom_q6v5_pas.c b/drivers/remoteproc/qcom_q6v5_pas.c
index 3153d8203..e7908dee5 100644
--- a/drivers/remoteproc/qcom_q6v5_pas.c
+++ b/drivers/remoteproc/qcom_q6v5_pas.c
@@ -1188,6 +1188,7 @@ static const struct of_device_id adsp_of_match[] = {
 	{ .compatible = "qcom,qcs404-adsp-pas", .data = &adsp_resource_init },
 	{ .compatible = "qcom,qcs404-cdsp-pas", .data = &cdsp_resource_init },
 	{ .compatible = "qcom,qcs404-wcss-pas", .data = &wcss_resource_init },
+	{ .compatible = "qcom,sc7180-adsp-pas", .data = &sm8250_adsp_resource},
 	{ .compatible = "qcom,sc7180-mpss-pas", .data = &mpss_resource_init},
 	{ .compatible = "qcom,sc7280-mpss-pas", .data = &mpss_resource_init},
 	{ .compatible = "qcom,sc8180x-adsp-pas", .data = &sm8150_adsp_resource},
diff --git a/sound/soc/qcom/sm8250.c b/sound/soc/qcom/sm8250.c
index 9626a9ef7..273c122a7 100644
--- a/sound/soc/qcom/sm8250.c
+++ b/sound/soc/qcom/sm8250.c
@@ -55,6 +55,14 @@ static int sm8250_snd_startup(struct snd_pcm_substream *substream)
 	struct snd_soc_dai *codec_dai = asoc_rtd_to_codec(rtd, 0);
 
 	switch (cpu_dai->id) {
+	case PRIMARY_MI2S_RX:
+		codec_dai_fmt |= SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_I2S;
+		snd_soc_dai_set_sysclk(cpu_dai,
+			Q6AFE_LPASS_CLK_ID_PRI_MI2S_IBIT,
+			MI2S_BCLK_RATE, SNDRV_PCM_STREAM_PLAYBACK);
+		snd_soc_dai_set_fmt(cpu_dai, fmt);
+		snd_soc_dai_set_fmt(codec_dai, codec_dai_fmt);
+		break;
 	case TERTIARY_MI2S_RX:
 		codec_dai_fmt |= SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_I2S;
 		snd_soc_dai_set_sysclk(cpu_dai,
